src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -9,11 +9,144 @@
 
 package common
 
-import "fmt"
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/container/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"golang.org/x/sync/errgroup"
+)
 
 const FeedbackMsg = `We'd love to hear about your experience with redpanda:
 https://vectorized.io/feedback`
 
 func DeprecationMessage(altCmd string) string {
 	return fmt.Sprintf("use '%s' instead.", altCmd)
 }
+
+// Try to read the config from the default expected locations, or from the
+// specific path passed with --config. If --config wasn't passed, and the config
+// wasn't found, return the default configuration.
+func FindConfigFile(
+	mgr config.Manager, configFile *string,
+) func() (*config.Config, error) {
+	var conf *config.Config
+	var err error
+	return func() (*config.Config, error) {
+		if conf != nil {
+			return conf, nil
+		}
+		conf, err = mgr.ReadOrFind(*configFile)
+		if err != nil {
+			log.Debug(err)
+			if os.IsNotExist(err) && *configFile == "" {
+				log.Debug(
+					"Config file not found and --config" +
+						" wasn't passed, using default" +
+						" config",
+				)
+				return config.Default(), nil
+			}
+		}
+		return conf, err
+	}
+}
+
+func ContainerBrokers() []string {
+	c, err := common.NewDockerClient()
+	if err != nil {
+		log.Debug(err)
+		return []string{}
+	}
+	nodes, err := common.GetExistingNodes(c)
+	if err != nil {
+		log.Debug(err)
+		return []string{}
+	}
+	grp := errgroup.Group{}
+	mu := sync.Mutex{}
+	addrs := []string{}
+	for _, node := range nodes {
+		s := node
+		grp.Go(func() error {
+			mu.Lock()
+			defer mu.Unlock()
+			addrs = append(
+				addrs,
+				fmt.Sprintf(
+					"127.0.0.1:%d",
+					s.HostKafkaPort,
+				),
+			)
+			return nil
+		})
+	}
+	err = grp.Wait()
+	if err != nil {
+		log.Debug(err)
```

Comment:
log.error? if there exists an error

Reply:
It's not fatal so it's ok to just debug it.

Reply:
Makes sense now to me